### PR TITLE
v0.1.0

### PR DESCRIPTION
**Pull request by betzrhodes**
Initial beta library release

### CODE REVIEW

**Comment by betzrhodes** [comment]
Tests are passing locally

**Comment by betzrhodes** [comment]
Could improve test code structure, but not worth holding up library release for.

**Comment by betzrhodes** [comment]
Automated testing set up - impFarm device A.

File: Scheduler.lib.nut [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+    // Start a new timer to trigger at a specific time
+    //
+    // Parameters:
+    //     t  (integer)     the time that the timer should fire
+    //     cb (function)    the function to run when the timer fires
+    // Return: (integer) the id number of the timer (can be used to cancel the timer)
+    function at(t, cb, ...) {
+        local now = date();
+        local hwSec = null;
+        if (_env != ENVIRONMENT_AGENT) hwSec = hardware.millis() / 1000.0;
```
Comment (ppetrosh) [inline Scheduler.lib.nut]: I would move this under the if/else statement below:

```
local hwSec = hardware.millis() / 1000.0;
```

This helps to make it unconditional, smaller and faster.

File: Scheduler.lib.nut [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+
+        local newJob = Scheduler.Job(this, {
+            "type": JOB_TYPE_REPEAT_FROM,
+            "id": _nextId++,
+            "repeat": int,
+            "cb": cb,
+            "args": vargv
+        });
+
+        if (_env == ENVIRONMENT_AGENT) {
```
Comment (ppetrosh) [inline Scheduler.lib.nut]: Why is it different for device and agent?

Comment (betzrhodes) [reply Scheduler.lib.nut]: Agent uses date usec for millisecond timing and and device uses hardware.millis.

**Comment by betzrhodes** [comment]
Working on a refactor (adding comments), so it is easier to maintain. Also cleaning up tests.

**Comment by betzrhodes** [comment]
Code refactored and comments added.
Simplified library. Removed Job methods: reset, pause, and unpause. 
Added Job status property and getStatus method. 
Updated Job now method.

**Pull request merged**